Update emscripten and XNNPack to the latest version

jinjingforever opened the pull request:
- Updated WORKSPACE and toolchain configs to make the build work with the new versions.

- The generated JS file from the new emscirpten changes how wasm module is loaded. Before, we call a factory function and it returns the initialized module. Now, the factory function returns a promise of the module. This requires some changes to backend_wasm.ts file.

- Other related updates for docs and scripts. See code comments for more info.

The new version of xnnpack+emscripten seems to fix the bug I saw about binary operator. It also fixes the frozen blazeface demo. I also tested some simple cases that use minified wasm backend bundle. 

Fixes #4710
Fixes #4525
Fixes #4311

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/tensorflow/tfjs/4730)
<!-- Reviewable:end -->

lina128 reviewed (approved):
The documentation part is great, and thank you for fixing this, Jing!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4730#-:-MUFYJGV46vh4ST1qckm:byuxmih)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @jinjingforever)

___
*[tfjs-backend-wasm/README.md, line 215 at r1](https://reviewable.io/reviews/tensorflow/tfjs/4730#-MUFY6AS8prJQuA1RTug:-MUFY6ATAVokWmZUuACp:bni1cei) ([raw file](https://github.com/tensorflow/tfjs/blob/d26e2644bc6790afbebb87c20bbc07a28d81e81a/tfjs-backend-wasm/README.md#L215)):*
> ```Markdown
> ## Emscripten installation
> 
> Install the Emscripten SDK (version 2.0.14):
> ```

We also need to update emscripten version in docker: https://github.com/tensorflow/tfjs/blob/master/dockers/wasm-docker/Dockerfile


<!-- Sent from Reviewable.io -->

jinjingforever commented:
Hi Na, I updated Pack.ts (which I think was the main culprit that caused the memory issue) and Concat.ts. Please help me double check:) Also updated the dock file. Let's see how it goes on CI.

Thanks again for your help!

lina128 reviewed (approved):
Thank you for fixing the memory leak issue, Jing! Amazing work finding and fixing two important bugs.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/4730#-:-MUGXge53y78IRrigjza:bhaplkv)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @jinjingforever)

___
*[tfjs-backend-wasm/src/kernels/Concat.ts, line 81 at r2](https://reviewable.io/reviews/tensorflow/tfjs/4730#-MUGXOLJ9ihgny3lMVKY:-MUGXOLJ9ihgny3lMVKZ:b-73m6vw) ([raw file](https://github.com/tensorflow/tfjs/blob/d31b525d3f00def291d0cdf7d42b1a197b56c9dd/tfjs-backend-wasm/src/kernels/Concat.ts#L81)):*
> ```XML
>     outData.stringBytes = backend_util.fromStringArrayToUint8(outVals);
> 
>     inputs2D.forEach(t => backend.disposeData(t.dataId));
> ```

Nice catch!


<!-- Sent from Reviewable.io -->
